internal/post/sender.go

```diff
@@ -35,22 +36,27 @@ func (sender *SMTPSender) FromAddress() Address {
 // SendEmail sends email message to the given recipient
 func (sender *SMTPSender) SendEmail(ctx context.Context, msg *Message) (err error) {
 	defer mon.Task()(&ctx)(&err)
-	// TODO: validate address before initializing SMTPSender
-	// suppress error because address should be validated
-	// before creating SMTPSender
-	host, _, _ := net.SplitHostPort(sender.ServerAddress)
 
 	client, err := smtp.Dial(sender.ServerAddress)
 	if err != nil {
 		return err
 	}
-	// close underlying connection
-	defer func() {
-		err = errs.Combine(err, client.Close())
-	}()
+
+	if err = sender.communicate(ctx, client, msg); err != nil {
+		return errs.Combine(err, client.Close())
+	}
+
+	return nil
```

Comment:
keep the `client.Quit` here, that way it's more clear that it gets closed, also that way you can remove write data and keep it inside

Reply:
Then I need to remove `StartTls` and `Auth` out from there, so then I can possibly reuse communicate method on open connection, because calling `client.Quit` here makes me think that I can still use the `client` before quitting it. Another reason is that if `client.Quit` returns an`error` than `client.Close` should be called explicitly  

`writedata` makes sure that `writer` is closed immediately  after it is used, also it makes `error` more clear as `writer.Close` error will be combined only with `writer.Write` error